src/VisualStudio/Xaml/Impl/Diagnostics/Analyzers/XamlDocumentDiagnosticAnalyzer.cs

```diff
@@ -17,30 +15,15 @@ namespace Microsoft.CodeAnalysis.Xaml.Diagnostics.Analyzers;
 internal sealed class XamlDocumentDiagnosticAnalyzer : DocumentDiagnosticAnalyzer
 {
     public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
-    {
-        get
-        {
-            return XamlProjectService.AnalyzerService?.SupportedDiagnostics ?? [];
-        }
-    }
-
-    public override async Task<ImmutableArray<Diagnostic>> AnalyzeSyntaxAsync(Document document, CancellationToken cancellationToken)
-    {
-        if (XamlProjectService.AnalyzerService == null)
-        {
-            return [];
-        }
-
-        return await XamlProjectService.AnalyzerService.AnalyzeSyntaxAsync(document, cancellationToken).ConfigureAwait(false);
-    }
+        => XamlProjectService.AnalyzerService?.SupportedDiagnostics ?? [];
 
-    public override async Task<ImmutableArray<Diagnostic>> AnalyzeSemanticsAsync(Document document, CancellationToken cancellationToken)
-    {
-        if (XamlProjectService.AnalyzerService == null)
-        {
-            return [];
-        }
+    public override async Task<ImmutableArray<Diagnostic>> AnalyzeSyntaxAsync(TextDocument textDocument, SyntaxTree? tree, CancellationToken cancellationToken)
+        => XamlProjectService.AnalyzerService is null || textDocument is not Document document
+            ? []
+            : await XamlProjectService.AnalyzerService.AnalyzeSyntaxAsync(document, cancellationToken).ConfigureAwait(false);
```

Comment:
>await [](http://example.com/codeflow?start=14&length=5)

Does this need to be awaited?

Reply:
yes.  because on different paths we return different values.

Reply:
Making sure we're thinking of the same change:
```
    public override Task<ImmutableArray<Diagnostic>> AnalyzeSyntaxAsync(TextDocument textDocument, SyntaxTree? tree, CancellationToken cancellationToken)
        => XamlProjectService.AnalyzerService is null || textDocument is not Document document
            ? SpecializedTasks.EmptyImmutableArray<Diagnostic>()
            : XamlProjectService.AnalyzerService.AnalyzeSyntaxAsync(document, cancellationToken);
```

Reply:
@CyrusNajmabadi  -- not a deal breaker, just curious why the suggested code wouldn't work

Reply:
That would work...  Or just the code already there :-)

Reply:
I'm not huge on avoiding async/await personally.